File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        if(heights.length == 0) {
            return 0;
        }

        int left = 0;
        int right = heights.length-1;
        int area = 0;

        while (left < right) {
            int width = right - left;
            int height = Math.min(heights[left], heights[right]);
            int tempA = height*width;
            area = Math.max(tempA, area);

            if(heights[left] < heights[right]) {
                left++;
            } else{
                right--;
            }

        }
        
        return area;
    }
}

File: Data Structures & Algorithms/two-sum-less-than-k/submission-2.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()

        closestSum= -1
        for left in range(len(nums)):
            for right in range(left+1, len(nums)):
                sum = nums[left] +nums[right]
                if sum < k:
                    closestSum = max(closestSum, sum)
        
        return closestSum

        
        